tensorflow/lite/kernels/internal/reference/reference_ops.h

```diff
@@ -2933,6 +2933,22 @@ inline void Cast(const RuntimeShape& input_shape, const SrcT* input_data,
   }
 }
 
+template <typename T>
+T FloorMod(T input1, T input2) {
+  struct FloatMod {
+    float operator()(const float lhs, const float rhs) const {
+      return std::fmod(lhs, rhs);
+    }
+  };
+  using ModFunc = typename std::conditional<std::is_integral<T>::value,
+                                            std::modulus<T>, FloatMod>::type;
+  ModFunc mod_func;
+  T trunc_mod = mod_func(input1, input2);
+  return (input1 < T(0)) == (input2 < T(0))
```

Comment:
Why is this different from the original one?

Reply:
@haozha111 , thanks for the review i have updated the code as per your comments, kindly check and approve.

Regards
Amit